### ISSUE TITLE
Regression using custom scalars with schema scoping.

### ISSUE THREAD

**Issue by coady** [strawberry-graphql/strawberry#1204]
v0.75 changed the scope of custom scalars from global to schema specific, but it's caused a subtle regression. Using the given example:
```python
    EpochDateTime = strawberry.scalar(
        datetime,
        serialize=lambda value: int(value.timestamp()),
        parse_value=lambda value: datetime.fromtimestamp(int(value), timezone.utc),
    )
```
the user could always `EpochDateTime` as an annotation, but the point of the example was to override `datetime` behavior - still using `datetime` as the annotation.

The problem now is using builtins _only_ works for overrides. I think it was implied that using an existing type should work, and would be more Pythonic as well. The following example fails in a surprising way:
```python
Long = strawberry.scalar(NewType("Long", int), description="64-bit int")
Binary = strawberry.scalar(
    bytes,
    name="Binary",
    serialize=lambda b: base64.b64encode(b).decode("utf8"),
    parse_value=base64.b64decode,
)
Duration = strawberry.scalar(
    timedelta,
    name="Duration",
    serialize=timedelta.total_seconds,
    parse_value=lambda s: timedelta(seconds=s),
)

@strawberry.type
class Query:
    @strawberry.field
    def long(self, value: Long) -> Long:
        return value

    @strawberry.field
    def binary(self, value: bytes) -> bytes:
        return value

    @strawberry.field
    def duration(self, value: timedelta) -> timedelta:
        return value

schema = strawberry.Schema(Query, scalar_overrides={bytes: Binary, timedelta: Duration})
```
* `Long` works - because it is the scalar
* `bytes` works accidentally - because it happens to be listed as a scalar in a global `SCALAR_TYPES` registry
* `timedelta` fails - because it's not an "override" (but neither is `bytes`)

Basically the schema converter's `scalar_registry` isn't being used as the source of truth on what is-a scalar.

**Comment by jkimbo** [comment]
Thanks for the bug report @coady . I would expect both cases to work, is that what you're expecting?

**Comment by jkimbo** [comment]
@coady I can't seem to reproduce part of your example. This fails for me:

```python
@strawberry.type
class Query:
    @strawberry.field
    def binary(self, value: bytes) -> bytes:
        return value

schema = strawberry.Schema(Query)
```

**Comment by jkimbo** [comment]
Ok nvm, I think I see the issue. I'll try and come up with a fix tomorrow.

**Comment by jkimbo** [comment]
@coady #1212 should fix this issue.
